haskell-src/Concordium/Types/Queries/Tokens.hs

```diff
@@ -39,3 +68,82 @@ data TokenAccountState = TokenAccountState
       memberDenyList :: !Bool
     }
     deriving (Eq, Show)
+
+-- | Parameter for a Token module.
+newtype TokenParameter = TokenParameter {parameterBytes :: BS.ShortByteString}
+    deriving (Eq)
+    deriving (Aeson.ToJSON, Aeson.FromJSON, Show) via ByteStringHelpers.ByteStringHex
+
+instance Serialize.Serialize TokenParameter where
+    get = do
+        len <- Serialize.getWord16be
+        TokenParameter <$> Serialize.getShortByteString (fromIntegral len)
+    put (TokenParameter parameter) = do
+        Serialize.putWord16be (fromIntegral (BS.length parameter))
+        Serialize.putShortByteString parameter
+
+-- | Module reference for a token module.
+newtype TokenModuleRef = TokenModuleRef {tokenModuleRef :: SHA256.Hash}
+    deriving (Eq, Ord, Hashable.Hashable)
+    deriving (Aeson.FromJSON, Aeson.ToJSON, Read) via SHA256.Hash
+
+instance Show TokenModuleRef where
+    show (TokenModuleRef m) = show m
```

Comment:
Is it not possible to derive the show instance?

Reply:
Yes you are right, `Serialize` as well actually.